lib/pure/json.nim

```diff
@@ -492,23 +648,22 @@ proc `[]`*(node: JsonNode, index: int): JsonNode {.inline.} =
   ## Gets the node at `index` in an Array. Result is undefined if `index`
   ## is out of bounds, but as long as array bound checks are enabled it will
   ## result in an exception.
-  assert(not isNil(node))
-  assert(node.kind == JArray)
+  if node.isNil or node.kind != JArray: return newJNull()
```

Comment:
This is specifically not what the semantics of `[]` should be. This is why we have ``{}``, that is the operator that is supposed to handle `nil`s like this.